### DISCUSSION TITLE
Unable to read file from Next.js API route deployed on Vercel

### DISCUSSION THREAD

**Discussion by dale-french** [vercel/community#949] [Help]
Following [this](https://vercel.com/guides/loading-static-file-nextjs-api-route) guide, I am unable to retrieve files listed in a directory from within an API route deployed on Vercel.

**Sample request:** 
`curl -XGET 'https://sales-funnel-s00wbw7zv-aynjil.vercel.app/api/test/testendpoint'`

**The code inside the API route (/pages/api/test/testendpoint):**
```
import path from 'path'
import {promises as fs} from 'fs'

export default async function handler(req, res) {
  try {
    //Find the absolute path of the json directory
    const jsonDirectory = path.join(process.cwd(), 'json')
    //Read the json data file data.json
    const fileContents = await fs.readFile(jsonDirectory + '/data.json', 'utf8')
    //Return the content of the data file in json format
    res.status(200).json({
      success: true,
      fileContents,
    })
  } catch (e) {
    res.status(500).json({
      success: false,
      error: e,
    })
  }
}

```

**Response:**
```
{
    "success": false,
    "error": {
        "errno": -2,
        "code": "ENOENT",
        "syscall": "open",
        "path": "/var/task/json/data.json"
    }
}
```

**Comment by amyegan** [comment]
Hi @dale-french! Are you able to share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look at the problem?

**Comment by Alexis-Bize** [comment]
I second that. I didn't have the issue two days ago and now I'm unable to read local files from an API route.

After digging a little bit, I noticed that the `.next/server/pages/api/[filePath]/[fileName].nft.json` won't include my `/resources` directory and its files if... There's too much lines of code (or imported modules / dependencies). After removing a few it worked.

~~Looks more like a Next.js issue here (at least for my case).~~
Erf. No, my recent Sentry installation is the issue I guess.

![Capture d’écran 2022-10-12 à 23 25 43](https://user-images.githubusercontent.com/1816892/195450957-f93d5ddc-31ae-4105-ad29-b0e741bebc69.png)

Managed to fix it by creating a dedicated file which exposes the path of my `/resources` folder and imported it into my API route.

```javascript
import { join } from 'path';
const RESOURCES_PATH = join(process.cwd(), 'resources');
export const computeResourcesPath = (path?: string) => join(RESOURCES_PATH, path);
```

@dale-french - Make sure that your /json folder is at the root of your project (not under src nor public).

**Comment by qodesmith** [comment]
@dale-french Have you tried joining the path from the root of the directory? So this instead:

```javascript
const jsonDirectory = path.join(process.cwd(), '/pages/api/json')
```

This is of course assuming that's the directory location relative to your project root - `<project root>/pages/api/json`.
